templates/terraform/constants/subscription.go.erb

```diff
@@ -24,3 +24,15 @@ func comparePubsubSubscriptionExpirationPolicy(_, old, new string, _ *schema.Res
 	}
 	return trimmedNew == trimmedOld
 }
+
+func comparePubsubSubscriptionRetryPolicy(_, old, new string, _ *schema.ResourceData) bool {
+	trimmedNew := strings.TrimLeft(new, "0")
+	trimmedOld := strings.TrimLeft(old, "0")
+	if strings.Contains(trimmedNew, ".") {
+		trimmedNew = strings.TrimRight(strings.TrimSuffix(trimmedNew, "s"), "0") + "s"
```

Comment:
though clever, this takes the reader a bit of effort to figure out what this does. What if you removed the 's' from each and then converted the rest to numbers? That feels like it accomplishes the same thing as this, but is a bit easier to understand.

I'm also curious in what case this would be necessary. Does the API support sending a request for this field without the 's' at the end? Does it ever return a number with extra leading/trailing zeros?